src/AutoRest.CSharp/Common/Input/InputTypes/Examples/Serialization/TypeSpecInputExampleValueConverter.cs

```diff
@@ -116,6 +118,69 @@ private InputExampleValue CreateObjectExample(ref Utf8JsonReader reader, string?
             return result;
         }
 
+        private InputExampleValue CreateUnknownExample(ref Utf8JsonReader reader, string? id, JsonSerializerOptions options, ReferenceResolver referenceResolver)
+        {
+            bool isFirstProperty = id == null;
+            InputType? type = null;
+            JsonElement? rawValue = null;
+            while (reader.TokenType != JsonTokenType.EndObject)
+            {
+                var isKnownProperty = reader.TryReadReferenceId(ref isFirstProperty, ref id)
+                    || reader.TryReadWithConverter(TypePropertyName, options, ref type)
+                    || reader.TryReadWithConverter(ValuePropertyName, options, ref rawValue);
+
+                if (!isKnownProperty)
+                {
+                    reader.SkipProperty();
+                }
+            }
+
+            var result = ParseUnknownValue(type ?? throw new JsonException(), rawValue);
+
+            if (id != null)
+            {
+                referenceResolver.AddReference(id, result);
+            }
+
+            return result;
+
+            static InputExampleValue ParseUnknownValue(InputType type, JsonElement? rawValue)
+            {
+                switch (rawValue?.ValueKind)
```

Comment:
This seems redundant with https://github.com/Azure/autorest.csharp/blob/725dad70cc367e9747ce2d6e088aaef6b6077b2e/src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputLiteralTypeConverter.cs#L52-L66.
Should we handle array and object in above logic as well? If not, what's the reason behind?

Is it possible to consolidate the logic to a helper method or a converter for unknown kind?